reference-implementation/lib/writable-stream.js

```diff
@@ -656,20 +650,26 @@ function WritableStreamDefaultWriterWrite(writer, chunk) {
 
   assert(stream !== undefined);
 
+  const controller = stream._writableStreamController;
+
+  const chunkSize = WritableStreamDefaultControllerGetChunkSize(controller, chunk);
```

Comment:
It's really unfortunate that we need to do this in the Writer's abstract op. Ideally, the detail of the operation should be hidden. But ok for now.